symengine/derivative.cpp

```diff
@@ -590,4 +591,20 @@ class DiffImplementation
 #include "symengine/type_codes.inc"
 #undef SYMENGINE_ENUM
 
+RCP<const Basic> diff(const RCP<const Basic> &arg, const RCP<const Symbol> &x) {
+    return arg->diff(x);
+}
+
+//! SymPy style differentiation for non-symbol variables
+// Since SymPy's differentiation makes no sense mathematically, it is
+// defined separately here for compatibility
```

Comment:
Difference is mentioned here. Basically `diff` takes only a `Symbol` as the w.r.t argument. `sdiff` takes non-symbols as well and differentiate it as in SymPy. (Substitute with a dummy symbol, differentiate and then substitute back)